src/lib/support/jsontlv/JsonToTlv.cpp

```diff
@@ -0,0 +1,344 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/support/Base64.h>
+#include <lib/support/SafeInt.h>
+#include <lib/support/jsontlv/JsonToTlv.h>
+#include <lib/support/jsontlv/JsonTypes.h>
+#include <nlohmann/json.hpp>
+#include <sstream>
+
+using json = nlohmann::ordered_json;
+
+namespace chip {
+
+namespace {
+
+std::vector<std::string> splitString(const std::string & input, char separator)
```

Comment:
`splitIntoLines` or similar to make it clear this is splitting by line?

Reply:
renamed: `splitString() --> splitIntoFieldsBySeparator()`